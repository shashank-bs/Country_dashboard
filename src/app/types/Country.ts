export interface Country {
    name: {
      common: string;
      official:string;
    };
    population: number;
    region: string;
    capital: string[];
    flags: {
      svg: string;
    };
    currencies: Record<string, { name: string }>;
    languages: Record<string, string>;
    timezones: string[];
    cca2:string;
  }
  