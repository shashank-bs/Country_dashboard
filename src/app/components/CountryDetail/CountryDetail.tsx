// app/components/CountryDetail.tsx
import { Typography, Box } from '@mui/material';
import { Country } from '../../types/Country';

interface CountryDetailProps {
  country: Country;
}

const CountryDetail: React.FC<CountryDetailProps> = ({ country }) => {
  return (
    <Box p={4}>
      <Typography variant="h4">{country.name.common}</Typography>
      <Typography variant="h6">Capital: {country.capital?.[0]}</Typography>
      <Typography variant="body1">Population: {country.population.toLocaleString()}</Typography>
      <Typography variant="body1">Region: {country.region}</Typography>
      <Typography variant="body1">Timezones: {country.timezones.join(', ')}</Typography>
      <Typography variant="body1">
        Languages: {Object.values(country.languages).join(', ')}
      </Typography>
      <Typography variant="body1">
        Currencies: {Object.values(country.currencies).map(c => c.name).join(', ')}
      </Typography>
    </Box>
  );
};

export default CountryDetail;
