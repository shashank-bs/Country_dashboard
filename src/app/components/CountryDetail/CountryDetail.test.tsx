// app/components/CountryDetail.test.tsx
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CountryDetail from "./CountryDetail";
import { Country } from "../../types/Country";

describe("CountryDetail Component", () => {
  const mockCountry: Country = {
    name: {
      common: "United States",
      official: "",
    },
    flags: { svg: "https://example.com/flag.svg" },
    capital: ["Washington, D.C."],
    population: 331000000,
    region: "Americas",
    currencies: {
      USD: { name: "United States dollar" },
    },
    languages: {
      eng: "English",
    },
    timezones: ["UTC-05:00"],
    cca2: "",
  };

  test("renders country name correctly", () => {
    const { getByText } = render(<CountryDetail country={mockCountry} />);
    expect(getByText("United States")).toBeInTheDocument();
  });

  test("renders country capital correctly", () => {
    const { getByText } = render(<CountryDetail country={mockCountry} />);
    expect(getByText("Capital: Washington, D.C.")).toBeInTheDocument();
  });

  test("renders country region correctly", () => {
    const { getByText } = render(<CountryDetail country={mockCountry} />);
    expect(getByText("Region: Americas")).toBeInTheDocument();
  });

  test("renders country timezones correctly", () => {
    const { getByText } = render(<CountryDetail country={mockCountry} />);
    expect(getByText("Timezones: UTC-05:00")).toBeInTheDocument();
  });

  test("renders country languages correctly", () => {
    const { getByText } = render(<CountryDetail country={mockCountry} />);
    expect(getByText("Languages: English")).toBeInTheDocument();
  });

  test("renders country currencies correctly", () => {
    const { getByText } = render(<CountryDetail country={mockCountry} />);
    expect(getByText("Currencies: United States dollar")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<CountryDetail country={mockCountry} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
