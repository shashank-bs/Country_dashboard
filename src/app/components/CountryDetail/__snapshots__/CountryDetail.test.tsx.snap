// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetail Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1wcaknn"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
    >
      United States
    </h4>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
    >
      Capital: Washington, D.C.
    </h6>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    >
      Population: 33,10,00,000
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    >
      Region: Americas
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    >
      Timezones: UTC-05:00
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    >
      Languages: English
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    >
      Currencies: United States dollar
    </p>
  </div>
</DocumentFragment>
`;
